src/test/java/org/junit/runners/CustomBlockJUnit4ClassRunnerTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.junit.runners;
+
+import static org.junit.Assert.*;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.Description;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+
+/**
+ * Tests that verify proper behavior for custom runners that extend
+ * {@link BlockJUnit4ClassRunner}.
+ *
+ * @author Sam Brannen
+ * @since 4.13
+ */
+public class CustomBlockJUnit4ClassRunnerTest {
+
+	/**
+	 * This code has been adapted from the Spring Framework.
+	 */
+	@Test
+	public void exceptionsFromMethodBlockMustNotResultInUnrootedTests() throws Exception {
+		Class<CustomBlockJUnit4ClassRunnerTestCase> testClass = CustomBlockJUnit4ClassRunnerTestCase.class;
+		TrackingRunListener listener = new TrackingRunListener();
+		RunNotifier notifier = new RunNotifier();
+		notifier.addListener(listener);
+
+		new CustomBlockJUnit4ClassRunner(testClass).run(notifier);
+		assertEquals("Number of tests started for [" + testClass + "].", 2, listener.getTestStartedCount());
+		assertEquals("Number of tests ignored for [" + testClass + "].", 0, listener.getTestIgnoredCount());
+		assertEquals("Number of assumption failures for [" + testClass + "].", 0, listener.getTestAssumptionFailureCount());
+		assertEquals("Number of test failures for [" + testClass + "].", 1, listener.getTestFailureCount());
+		assertEquals("Number of tests finished for [" + testClass + "].", 2, listener.getTestFinishedCount());
+	}
+
+
+	@Ignore("This test case is run manually by the enclosing test class")
```

Comment:
Define "need". ;)

When executing tests via `org.junit.tests.AllTests`, you are correct: the `@Ignore` is unnecessary.

However, when executing tests within an IDE such as Eclipse, it is very useful to have such _internal_ test classes disabled by default. In that manner, there are no false negatives when executing all tests within a given package, class, etc. Otherwise, such tests that always fail (intentionally) will produce undesirable noise and distract developers from the task at hand.